packages/frontend/amp/components/elements/SoundcloudBlockComponent.tsx

```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+
+export const SoundcloudBlockComponent: React.SFC<{
+    element: SoundcloudBlockElement;
+}> = ({ element }) => {
+    if (element.playlist == null && element.track == null) {
+        if (element.isMandatory) {
+            throw new Error(
```

Comment:
What do you think is the right error-handling behaviour here?

I'd be tempted to say something like: if an element is absolutely required for the page, then error dramatically if it's not valid, but otherwise, log the error and continue. (This might happen at the Elements level though, rather than here.)

Reply:
Maybe this is a problem with the data type for the element as much as handling.

Reply:
Ah good point.

I'll write something about error handling anyway for our next weekly meeting or before as a discussion point more generally as it feels time to nail this down!